docs/docs/themes/shadowing.md

```diff
@@ -0,0 +1,215 @@
+---
+title: Shadowing
+---
+
+
+Gatsby themes introduce a concept called "shadowing". This feature allows users to override a component, or any other file, in the `src` directory to customize it.
+
+A practical use case is when you've installed `gatsby-theme-blog` and want to customize the author Bio component to add your own biographical content. Component shadowing lets you replace the theme’s original file, `gatsby-theme-blog/src/components/bio.js`, with your own file to override with any changes you need.
+
+## Shadowing Example
+
+If you’ve installed `gatsby-theme-blog` you’ll notice that it renders a `Bio` component which is used in the `BlogPost` template. If you’d like to change the `Bio` component you can do so with component shadowing.
+
+### Theme file structure
+
+You can look at the `gatsby-theme-blog`'s file structure to determine the file path for the file you want to shadow.
+
+```txt
+├── gatsby-config.js
+├── gatsby-node.js
+└── src
+    ├── components
+    │   ├── bio-content.js
+    │   ├── bio.js // highlight-line
+    │   ├── header.js
+    │   ├── headings.js
+    │   ├── home-footer.js
+    │   ├── layout.js
+    │   ├── post-footer.js
+    │   ├── post.js
+    │   ├── posts.js
+    │   ├── seo.js
+    │   └── switch.js
+    ├── gatsby-plugin-theme-ui
+    │   ├── colors.js
+    │   ├── components.js
+    │   ├── index.js
+    │   ├── prism.js
+    │   ├── styles.js
+    │   └── typography.js
+    └── templates
+        ├── post.js
+        └── posts.js
+```
+
+### Customizing the `Bio` component
+
+In this case, the file to shadow is `gatsby-theme-blog/src/components/bio.js`.
+
+Component shadowing relies on a mirrored file structure to determine which component will be rendered. In order to override the Bio component in `gatsby-theme-blog`, create a file named `user-site/src/gatsby-theme-blog/components/bio.js`.
+
+Any file that lives in the `src/gatsby-theme-blog` directory of the user’s site will be used instead of a file with the same name located in the theme’s src directory: `gatsby-theme-blog/src`.
+
+This means that `user-site/src/gatsby-theme-blog/components/bio.js` will be rendered in place of `gatsby-theme-blog/src/components/bio.js`:
+
+```js:title=src/gatsby-theme-blog/components/bio.js
+import React from "react"
+export default () => <h1>My new bio component!</h1>
+```
+
+A successful shadow of the Bio component will result in the following directory tree:
+
+```txt
+user-site
+└── src
+    └── gatsby-theme-blog
+        └── components
+            └── bio.js // highlight-line
+```
+
+## Shadowing parent themes
+
+Some themes, including `gatsby-theme-blog`, install other themes. `gatsby-theme-blog` uses `gatsby-plugin-theme-ui`. If you want to customize any of its implementation you can do so with shadowing as well.
+
+To do so, create a file named `user-site/src/gatsby-theme-blog/gatsby-plugin-theme-ui/index.js`.
```

Comment:
the `index.js` file in `gatsby-plugin-theme-ui` isn't a component, so we should use `file` to describe it.

> To shadow the `index.js` file from the parent theme `gatsby-plugin-theme-ui`,

---

also it occurs to me that the naming here is awkward because we're calling something with plugin in its name a theme ("the parent theme: gatsby-plugin-theme-ui"). Perhaps we should drop the "parent theme" nomenclature from this entire section since shadowing applies to all plugins now anyway and we don't reference "parent plugins" anywhere.

Reply:
I updated the suggestion to be shorter, great tips!